testlib/test_data.py

```diff
@@ -597,6 +597,49 @@ def test_update(self):
         gramex.data.update(data, args=args, id=['देश', 'city', 'product'])
         ase(types_original, data.dtypes)
 
+    def test_update_multiple_file(self):
+        # Test on a file
+
+        update_file = os.path.join(folder, 'actors.update.csv')
+        shutil.copy(os.path.join(folder, '..', 'tests/actors.csv'), update_file)
+        self.tmpfiles.append(update_file)
+
+        names = ['Humphrey Bogart', 'James Stewart', 'Audrey Hepburn']
+        categories = ['Stars', 'Thespians', 'Heartthrobs']
+        ratings = [1, 0.99, 1.11]
+        gramex.data.update(
+            update_file,
+            args={
+                'name': names,
+                'category': categories,
+                'rating': ratings
+            }, id=['name']
+        )
+        df = gramex.data.filter(update_file, args={'name': names})
+        self.assertEqual(df['category'].tolist(), categories)
+        self.assertEqual(df['rating'].tolist(), ratings)
+
+    def test_update_multiple_db(self):
+        actors = gramex.cache.open(os.path.join(folder, '../tests/actors.csv'))
+        temp_db = f'sqlite:///{folder}/actors.db'
+        self.tmpfiles.append(os.path.join(folder, 'actors.db'))
+        actors.to_sql('actors', sa.create_engine(temp_db), index=False)
+
+        names = ['Humphrey Bogart', 'James Stewart', 'Audrey Hepburn']
+        categories = ['Stars', 'Thespians', 'Heartthrobs']
+        ratings = [1, 0.99, 1.11]
```

Comment:
Please add a test case where 

- `ratings` has only 2 values [1, 0.99] and the third is missing. Audrey's rating should not get updated, but her categories should be.
- `names` has only 2 values -- in which case, the 3rd category and rating are ignored

Reply:
@sanand0 This works for files, but for sqlalchemy, to deal with uneven args, we might have to do column-wise update queries. Is this acceptable?

Reply:
@jaidevd -- yes, column-wise update queries are fine. I don't anticipate this to be used often and we can optimize later